### PR TITLE
Tiny performance improvements

### PR DESCRIPTION
**Pull request by dcousens**

### CODE REVIEW

File: browser.js [hunk 1]
```diff
@@ -78,9 +77,8 @@ function pbkdf2Sync (password, salt, iterations, keylen, digest) {
       }
     }
 
-    var destPos = (i - 1) * hLen
-    var len = (i === l ? r : hLen)
-    T.copy(DK, destPos, 0, len)
```
Comment (dcousens) [inline browser.js]: `var len` is probably important, but, none of our tests cover it actually doing anything useful.
So uh... ?

File: browser.js [hunk 2]
```diff
@@ -48,13 +48,13 @@ function pbkdf2Sync (password, salt, iterations, keylen, digest) {
...
   var hLen
   var l = 1
   var DK = new Buffer(keylen)
   var block1 = new Buffer(salt.length + 4)
   salt.copy(block1, 0, 0, salt.length)
 
-  var r
   var T
 
   for (var i = 1; i <= l; i++) {
```
Comment (fanatid) [inline browser.js]: What purpose of this `for-loop`? Why not unroll it?

Comment (dcousens) [inline browser.js]: @fanatid unfortunately I don't know a way to get the hash output size before entering the loop... the API's might have changed though.

**Comment by dcousens** [comment]
ping @calvinmetcalf any thoughts on this?

File: browser.js (hunk 1 above)
Comment (calvinmetcalf) [inline browser.js]: maybe if you specify a single iteration and a length that is longer or shorter then the hash algo ?

Comment (dcousens) [inline browser.js]: @calvinmetcalf should we add a test fixture for that?

**Comment by dcousens** [comment]
ping @calvinmetcalf or @fanatid ,  close or keep?

**Comment by calvinmetcalf** [comment]
happy to review once its rebased and passing the tests

**Comment by dcousens** [comment]
Tests are failing for an unrelated reason...

File: browser.js (hunk 1 above)
Comment (calvinmetcalf) [inline browser.js]: oo and can you add a fixture for this test ?

**Comment by dcousens** [comment]
Tests appear to be passing now... what happened there?

**Comment by calvinmetcalf** [comment]
coverage went down it looks like

**Comment by calvinmetcalf** [comment]
ok this plus #32 together 👍

**Comment by dcousens** [comment]
`npm test` fails for me locally on both `master` and `eh` still though.

``` bash
node -v
v6.2.2
```

**Comment by calvinmetcalf** [comment]
So they previously failed due to coverage being added will look into the
encoding changes in node 6

**Comment by dcousens** [comment]
Rebased and tests passing locally

**Comment by dcousens** [comment]
Almost all tests passing on Travis,  only failing because https://github.com/crypto-browserify/pbkdf2/issues/37.

@calvinmetcalf this must've been written this way for a reason?

**Pull request closed without merging**